scripts/update-ssh.sh

```diff
@@ -12,9 +12,17 @@ if test -f "keys/$user"; then
     cat "keys/$user" > "$tmp"
 fi
 
-# import current gh keys on top
-test "$RATELIMITED" == "true" || ssh-import-id "gh:$user" --output "$tmp"
+# workaround for gh ratelimits when runing tests locally
+cache=.ssh-key-cache/$user
+# if we have a cached file, and TEST is set, then use cache
+if test -f "$cache" -a -n "${TEST}"; then
+    cat "$cache" >> "$tmp"
+else
+    # add current gh keys into $tmp
+    ssh-import-id "gh:$user" --output "$tmp"
+fi
 
 # replace current authorized_keys
 mkdir -p "/home/$user/.ssh"
 mv "$tmp" "/home/$user/.ssh/authorized_keys"
+chown -R "$user:$user" "/home/$user/.ssh/"
```

Comment:
@bloodearnest do we also need to do e.g. `chmod 0600 /home/$user/.ssh/authorized_keys`?

Reply:
Sure - we could possibly do `chmod 0600 ./ssh/*`? 

I usually 600 the private key files, but not the config or pub keys. Is there a particular attack I'm not aware of with a world readable authorized keys?

Reply:
oh no, `0644` is also fine, just one of those files that I think better to be explicit about permissions than lean on the default. More worried about leaving extra write flags & sshd refusing to use it, than about a successful attack.